build.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::process::Command;
+
+// this build script is called on every build of firecracker,
+// after making a change in the source files,
+// embedding the FIRECRACKER_VERSION directly in the resulting binary
+fn main() {
+    let git_version_command = Command::new("git").args(&["describe", "--dirty"]).output();
+
+    match git_version_command {
+        Ok(outcome) => {
+            if outcome.status.success() {
+                println!(
+                    "cargo:rustc-env=FIRECRACKER_VERSION={}",
+                    String::from_utf8(outcome.stdout)
+                        .unwrap()
```

Comment:
I just realized this is another possible panic. In theory it shouldn't happen, but it would be better to handle it.